### DISCUSSION TITLE
OBOE call causes App to crash

### DISCUSSION THREAD

**Discussion by Ateros4me** [google/oboe#1448] [Q&A] [answered]
I have written the following code in Kotlin to set up an Audio Stream from audio in to audio out
class AudioActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_audio)

        // Call the final button testing routine
        val haAudioButton = findViewById<Button>(R.id.audioReturnButton)
        haAudioButton.setOnClickListener {
            haAudioRecordStart = false
            haDealWithAudioButton()
        }
	
        // Call function to print some info
        printInfo()

        // Thread to start Audio Engine
        haStartAudioEngine()
    }

    private fun haStartAudioEngine () {
        // Set the audio to urgent priority
        android.os.Process.setThreadPriority(android.os.Process.THREAD_PRIORITY_URGENT_AUDIO)

        // Start the threads
        NativeInterface.createAudioEngine()
        NativeInterface.enable(true)
    }

If I comment out the NativeInterface calls the App doesn't crash.
Can someone explain why the OBOE calls are crashing the App?
Note that the NativeInterface and associated CPP files are all copied from the fxlab example in the OBOE github library.

**Comment by philburk** [comment, marked as answer]
There is not enough information here for us to do anything. For a crash like this please use the Bug Report template at:

https://github.com/google/oboe/issues/new/choose

**Reply by Ateros4me** [reply]
I've added the build.gradle if that would help.
It's odd that in the Android Studio when I click on the native-lib file, it says that this file is not part of the project which it should be if compiling correctly.
The build.gradle is this

apply plugin: 'com.android.application'
apply plugin: 'kotlin-android'


android {
    buildToolsVersion '31.0.0'
    compileSdkVersion 31
    
    compileOptions {
        sourceCompatibility JavaVersion.VERSION_1_8
        targetCompatibility JavaVersion.VERSION_1_8
    }

    buildFeatures {
        prefab true
    }
    
    defaultConfig {
        applicationId "com.hearingaid"
        minSdkVersion 25
        targetSdkVersion 25
	
        versionCode 2
        versionName "1.0"

        testInstrumentationRunner "androidx.test.runner.AndroidJUnitRunner"
        externalNativeBuild {
            cmake {
              arguments "-DANDROID_STL=c++_shared"
                cppFlags ''
            }
        }
    }

    buildTypes {
        release {
            minifyEnabled false
            proguardFiles getDefaultProguardFile('proguard-android-optimize.txt'), 'proguard-rules.pro'
        }
    }
    kotlinOptions {
        jvmTarget = '1.8'
    }
    externalNativeBuild {
        cmake {
            path file('src/main/cpp/CMakeLists.txt')
            version '3.10.2'
        }
    }

}

dependencies {
    implementation fileTree(dir: 'libs', include: ['*.jar'])
    implementation 'androidx.core:core-ktx:1.6.0'
    implementation 'androidx.appcompat:appcompat:1.3.1'
    implementation 'com.google.android.material:material:1.4.0'
    implementation 'androidx.constraintlayout:constraintlayout:2.1.0'
    implementation 'com.google.oboe:oboe:1.6.1'
    testImplementation 'junit:junit:4.+'
    androidTestImplementation 'androidx.test.ext:junit:1.1.3'
    androidTestImplementation 'androidx.test.espresso:espresso-core:3.4.0'
}

**Reply by Ateros4me** [reply]
This is the CMakeLists.txt file

# For more information about using CMake with Android Studio, read the
# documentation: https://d.android.com/studio/projects/add-native-code.html

# Sets the minimum version of CMake required to build the native library.

cmake_minimum_required(VERSION 3.4.1)

# Require C++17 compiler with no extensions
set(CMAKE_CXX_STANDARD 17)
set(CMAKE_CXX_STANDARD_REQUIRED ON)
set(CMAKE_CXX_EXTENSIONS OFF)

link_directories(${CMAKE_CURRENT_LIST_DIR}/..)

# Creates and names a library, sets it as either STATIC
# or SHARED, and provides the relative paths to its source code.
# You can define multiple libraries, and CMake builds them for you.
# Gradle automatically packages shared libraries with your APK.

add_library( # Sets the name of the library.
        native-lib
        # Sets the library as a shared library.
        SHARED
        # Provides a relative path to your source file(s).
        src/main/cpp/native-lib.cpp
        src/main/cpp/DuplexEngine.cpp
        )

# Compile with warnings for safety
if(MSVC)
    target_compile_options(native-lib PRIVATE /W4 /WX)
else()
    target_compile_options(native-lib PRIVATE -Wall -Wextra -pedantic)
endif()

target_compile_options(native-lib PRIVATE -Ofast)

# Searches for a specified prebuilt library and stores the path as a
# variable. Because CMake includes system libraries in the search path by
# default, you only need to specify the name of the public NDK library
# you want to add. CMake verifies that the library exists before
# completing its build.

find_library( # Sets the name of the path variable.
        log-lib
        # Specifies the name of the NDK library that
        # you want CMake to locate.
        log)


### INCLUDE OBOE LIBRARY ###
find_package (oboe REQUIRED CONFIG)

### END OBOE INCLUDE SECTION ###

include_directories(src/main/cpp/)

# Specifies libraries CMake should link to your target library. You
# can link multiple libraries, such as libraries you define in this
# build script, prebuilt third-party libraries, or system libraries.

target_link_libraries( # Specifies the target library.
        native-lib
	oboe::oboe
	# Links the target library to the log library
        # included in the NDK.
        ${log-lib}
        )

**Reply by Ateros4me** [reply]
It seems to have interpreted the hash as a bold and chosen an increased size of text!
